ide/rust/sync_info_aspect.bzl

```diff
@@ -0,0 +1,179 @@
+#
```

Comment:
This is the **Cargo Sync Info Aspect,** which propagates through the Rust dependency tree via the `deps`, `proc_macro_deps` and `crate` attributes (which exist on some or all of `[rust_binary, rust_library, rust_test]`). It aggregates the crate names and versions (and Cargo `features`) of the Rust targets that it traverses, providing that information to the targets higher up in the dependency tree that depend on those targets.

In the end, we use that information to produce Sync Info in the form of `.ide-sync.properties` files in `bazel-bin` (e.g: `typeql_lang.ide-sync.properties`), which does nothing by itself, but the `RepoCargoManifestGenerator` analyses them and uses them to generate Cargo project structure in the form of `Cargo.toml` files.